File: sdks/node/src/modules/detection.ts

import { Client } from "../client";

export class DetectionClient {
  constructor(private client: Client) {}

  async detectImage(file: Buffer | Blob, filename: string = "image.jpg") {
    // In real implementation, this would use FormData
    return this.client.request("POST", "/detect/image", { file }, true);
  }

  async detectAudio(file: Buffer | Blob, filename: string = "audio.wav") {
    return this.client.request("POST", "/detect/audio", { file }, true);
  }

  async detectVideo(file: Buffer | Blob, filename: string = "video.mp4") {
    return this.client.request("POST", "/detect/video", { file }, true);
  }
}

File: sdks/python/fiduscan/detection.py

class DetectionClient:
    def __init__(self, client):
        self.client = client

    def detect_image(self, file_path):
        with open(file_path, "rb") as f:
            return self.client.request("POST", "/detect/image", files={"file": f})

    def detect_audio(self, file_path):
        with open(file_path, "rb") as f:
            return self.client.request("POST", "/detect/audio", files={"file": f})

    def detect_video(self, file_path):
        with open(file_path, "rb") as f:
            return self.client.request("POST", "/detect/video", files={"file": f})
